src/lib/sql/sql_translator.cpp

```diff
@@ -1722,8 +1761,35 @@ std::shared_ptr<AbstractExpression> SQLTranslator::_translate_hsql_expr(
     case hsql::kExprHint:
       FailInput("Hints are not yet supported");
 
-    case hsql::kExprCast:
-      FailInput("Explicit casts are not yet supported");
+    case hsql::kExprCast: {
+      const auto source_data_type = left->data_type();
+      if (expr.columnType.data_type == hsql::DataType::DATE) {
+        AssertInput(source_data_type == DataType::String, "Cannot cast " + left->as_column_name() + " as Date");
+        if (left->type == ExpressionType::Value) {
+          const auto date_string = std::string{boost::get<pmr_string>(static_cast<ValueExpression&>(*left).value)};
+          const auto date = string_to_date(date_string);
+          if (date) return left;
+          FailInput("'" + date_string + "' is not a valid date");
+        }
+        // We don't know if input actually contains dates, but maybe it is good enough that it cointains strings.
+        return left;
```

Comment:
Is this enough?
Pro: We don't really know dates at all, and we avoid expensive expression evaluator scans when a query requests `... cast(date_column as date) + 30 days`
Con: Does not actually check if all column values are dates, obviously